### PR TITLE
customize clear icon

### PR DESCRIPTION
**Pull request by buddhaCode**
I would like to change the clear button's icon. So I introduced a `clearIcon()` method.

### CODE REVIEW

**Comment by AjithLalps** [comment]
I have the same requirements and about to create a PR for it. Thank you @buddhaCode for the PR +1

**Comment by AjithLalps** [comment]
Small suggestion though, for me the function name `icon()` looks cleaner and readable than the name `clearIcon()`. Since we have only one icon for the widget I think it's safe to name the function as `icon()`. So we will have same naming like filament.

**Comment by buddhaCode** [comment]
@AjithLalps That sounds reasonable. I was checking if the picker component already inherits an `icon()` method from somewhere. That's not the case.
But after thinking about it, I found out that this Picker component uses the `HasAffixes` trait and simply uses the `suffixIcon()` method. So introducing a `clearIcon()`, as proposed in this PR, is more or less redundant.

I would suggest the following changes to this PR:

- remove `clearIcon()` from the `DateRangePicker`
- change `clearIcon()` to `suffixIcon()` on the `DateRangeFilter`

@AjithLalps and @malzariey What do you think about that?

**Comment by AjithLalps** [comment]
Agreed+1

File: src/Fields/DateRangePicker.php [hunk 1]
```diff
@@ -75,6 +76,18 @@ public function disableClear(bool|Closure $disable = true) : static
...
+    public function icon(string|Closure|null $icon = null): static
+    {
+        $this->icon = $icon;
+
+        return $this;
+    }
+
+    public function getIcon(): string
+    {
+        return $this->evaluate($this->icon) ?? 'heroicon-m-calendar-days';
```
Comment (AjithLalps) [inline src/Fields/DateRangePicker.php:88]: Can we set the default icon to the property?

**Comment by buddhaCode** [comment]
Hmm it wasn't that easy. When the button is deactivated is just the suffix icon. But when it's active is a suffix action which an icon.

So basically I left every thing as it was and just changed `clearIcon` to `icon` :)

**Review by AjithLalps — COMMENTED** [review]
I greatly appreciate your effort putting in this PR. I have a small feedback, thinking of making the code readable.

You can safely ignore this if it doesn't make sense.

Once again thank you for the effort.

File: src/Fields/DateRangePicker.php (hunk 1 above)
Comment (buddhaCode) [reply src/Fields/DateRangePicker.php:88]: I could do that for the picker. However, if I don't set an icon on the filter, the icon on the picker would be nulled.

Calling `DateRangeField::make()`would hand down a `null` icon to `DateRangePicker` in `src/Filters/DateRangeFilter.php:L196`.

But then I would also have to set the default icon for the filter, too. And then we would have the default icon on the picker and on the filter. That would be duplicate code again.

Comment (AjithLalps) [reply src/Fields/DateRangePicker.php:88]: Understood

**Comment by AjithLalps** [comment]
Thank you @malzariey for merging the PR and thank you @buddhaCode for the PR.

**Comment by buddhaCode** [comment]
@AjithLalps It was a pleasure. Thanks for your feedback.

**Comment by malzariey** [comment]
Thanks !

**Pull request merged**